### PR TITLE
test: Add integration test for envelope with transaction

### PR DESCRIPTION
**Pull request by flub**
This adds an integration test which inspects the envelope of a
transaction sent by the native SDK.

-------

### CODE REVIEW

File: tests/test_integration_http.py [hunk 1]
```diff
@@ -413,3 +415,60 @@ def delayed(req):
...
+    # assert trace_context['public_key']
+
+    assert trace_context["span_id"]
+    assert trace_context["status"] == "ok"
+
+    RFC3339_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+    start_timestamp = time.strptime(json["start_timestamp"], RFC3339_FORMAT)
+    assert start_timestamp
+    timestamp = time.strptime(json["timestamp"], RFC3339_FORMAT)
+    assert timestamp >= start_timestamp
```
Comment (Swatinem) [inline tests/test_integration_http.py:474]: I think this should be fine, as the example runs all these on the main thread, however we did run into the issue previously that our time source is not guaranteed to be fully monotonic in multithreaded scenarios :-(

**Review by Swatinem — APPROVED** [review]
❤️ This looks so much better now that someone familiar with pytest does these things ;-)

File: tests/test_integration_http.py (hunk 1 above)
Comment (flub) [reply tests/test_integration_http.py:474]: hum, yes i did worry about this.  if it turns out to be flaky i'll not hesitate the assert though

**Comment by flub** [comment]
> ❤️ This looks so much better now that someone familiar with pytest does these things ;-)

I did try and keep changes to some minimum to just make me debug things.  it could be a lot better but gets the job done.

File: tests/test_integration_http.py (hunk 1 above)
Comment (flub) [reply tests/test_integration_http.py:474]: huh "to remove the assert"

**Pull request merged**